diff --git a/data/Smarty/templates/admin/order/edit.tpl b/data/Smarty/templates/admin/order/edit.tpl
index 35a9b8c..c9af303 100644
--- a/data/Smarty/templates/admin/order/edit.tpl
+++ b/data/Smarty/templates/admin/order/edit.tpl
@@ -768,7 +768,14 @@
                     </select>
                 </td>
             </tr>
-
+            <!--{if $arrForm[$key].value == constant('SC_Helper_AmazonPay::PAYMENT_ID')}-->
+            <tr>
+                <th>AmazonリファレンスID</th>
+                <td>
+                    <a href="https://sellercentral.amazon.co.jp/external-payments/pmd/payment-details?orderReferenceId=<!--{$arrForm.memo05.value|h}-->" target="_blank"><!--{$arrForm.memo05.value|h}--></a>
+                </td>
+            </tr>
+            <!--{/if}-->
             <!--{if !empty($arrForm.payment_info)}-->
             <tr>
                 <th><!--{$arrForm.payment_type}-->情報</th>
diff --git a/data/Smarty/templates/default/cart/index.tpl b/data/Smarty/templates/default/cart/index.tpl
index 657b889..6a7c187 100644
--- a/data/Smarty/templates/default/cart/index.tpl
+++ b/data/Smarty/templates/default/cart/index.tpl
@@ -167,6 +167,9 @@
                                         <input type="image" class="hover_change_image" src="<!--{$TPL_URLPATH|h}-->img/button/btn_buystep.jpg" alt="購入手続きへ" name="confirm" />
                                     <!--{/if}-->
                                 </li>
+                                <li style="display:block; width:100%; padding-right: 0">
+                                    <div style="width:400px;height:45px; margin:10px auto" id="AmazonPayButton<!--{$key}-->"></div>
+                                </li>
                             </ul>
                         </div>
                     </form>
@@ -177,3 +180,4 @@
         <!--{/if}-->
     </div>
 </div>
+<!--{include $tpl_cart_button}-->
diff --git a/data/Smarty/templates/default/shopping/confirm.tpl b/data/Smarty/templates/default/shopping/confirm.tpl
index edc5ae1..4540f4a 100644
--- a/data/Smarty/templates/default/shopping/confirm.tpl
+++ b/data/Smarty/templates/default/shopping/confirm.tpl
@@ -60,7 +60,11 @@
                     </li>
                         <!--{else}-->
                     <li>
-                        <input type="image" onclick="return fnCheckSubmit();" class="hover_change_image" src="<!--{$TPL_URLPATH}-->img/button/btn_order_complete.jpg" alt="ご注文完了ページへ" name="next-top" id="next-top" />
+                        <!--{if $renderAmazonPayButton}-->
+                               <div style="width:400px;height:45px; margin:10px auto" id="AmazonPayButton<!--{$cartKey}-->"></div>
+                        <!--{else}-->
+                            <input type="image" onclick="return fnCheckSubmit();" class="hover_change_image" src="<!--{$TPL_URLPATH}-->img/button/btn_order_complete.jpg" alt="ご注文完了ページへ" name="next-top" id="next-top" />
+                        <!--{/if}-->
                     </li>
                     <!--{/if}-->
                 </ul>
@@ -238,6 +242,7 @@
             <!--{* ▼お届け先 *}-->
             <!--{foreach item=shippingItem from=$arrShipping name=shippingItem}-->
                 <h3>お届け先<!--{if $is_multiple}--><!--{$smarty.foreach.shippingItem.iteration}--><!--{/if}--></h3>
+                <!--{if !empty($arrForm['memo07'])}--> <p><a href="javascript:;" id="changeButton1"><strong>お届け先を変更する</strong></a></p><!--{/if}-->
                 <!--{if $is_multiple}-->
                     <table summary="ご注文内容確認">
                         <col width="10%" />
@@ -348,7 +353,7 @@
                 </tr>
                 <tr>
                     <th scope="row">お支払方法</th>
-                    <td><!--{$arrForm.payment_method|h}--></td>
+                    <td><!--{if !empty($arrForm['memo07'])}--> <!--{$arrForm['memo07']|h}--> <a href="javascript:;" id="changeButton2" style="color:#006aa6;"><i class="fa-solid fa-arrow-up-right-from-square"></i> お支払い方法を変更する</a><!--{else}--><!--{$arrForm.payment_method|h}--> <!--{/if}--></td>
                 </tr>
                 <tr>
                     <th scope="row">その他お問い合わせ</th>
@@ -376,3 +381,17 @@
         </form>
     </div>
 </div>
+<script src="https://static-fe.payments-amazon.com/checkout.js"></script>
+<script type="text/javascript" charset="utf-8">
+ amazon.Pay.bindChangeAction('#changeButton1', {
+     amazonCheckoutSessionId: '<!--{$arrForm['memo04']|h}-->',
+     changeAction: 'changeAddress'
+ });
+ amazon.Pay.bindChangeAction('#changeButton2', {
+     amazonCheckoutSessionId: '<!--{$arrForm['memo04']|h}-->',
+     changeAction: 'changePayment'
+ });
+</script>
+<!--{if $renderAmazonPayButton}-->
+    <!--{include $tpl_cart_button}-->
+<!--{/if}-->
diff --git a/data/Smarty/templates/default/shopping/index.tpl b/data/Smarty/templates/default/shopping/index.tpl
index 169d7b6..51b8117 100644
--- a/data/Smarty/templates/default/shopping/index.tpl
+++ b/data/Smarty/templates/default/shopping/index.tpl
@@ -19,7 +19,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
  *}-->
-
+<script src="https://static-fe.payments-amazon.com/checkout.js"></script>
 <div id="undercolumn">
     <div id="undercolumn_login">
         <h2 class="title"><!--{$tpl_title|h}--></h2>
@@ -28,6 +28,27 @@
             <input type="hidden" name="mode" value="login" />
 
             <div class="login_area">
+                <h3>Amazonのアカウントをお持ちのお客様</h3>
+                <div id="AmazonLoginButton"></div>
+                    <script>
+     amazon.Pay.renderButton('#AmazonLoginButton', {
+         // set checkout environment
+         merchantId: '<!--{$smarty.env.AMAZONPAY_MERCHANT_ID|h}-->',
+         ledgerCurrency: 'JPY',
+         sandbox: Boolean('<!--{$smarty.env.AMAZONPAY_SANDBOX|h}-->'),
+         // customize the buyer experience
+         checkoutLanguage: 'ja_JP',
+         productType: 'SignIn',
+         placement: 'Other',
+         buttonColor: 'Gold',
+         // configure Create Checkout Session request
+         signInConfig: {
+             payloadJSON: '<!--{$payload|json_encode}-->', // string generated in step 2
+             signature: '<!--{$signature}-->',
+             publicKeyId: '<!--{$smarty.env.AMAZONPAY_PUBLIC_KEY_ID|h}-->'
+         }
+     });
+    </script>
                 <h3>会員登録がお済みのお客様</h3>
                 <p class="inputtext">会員の方は、登録時に入力されたメールアドレスとパスワードでログインしてください。</p>
                 <div class="inputbox">
diff --git a/data/Smarty/templates/default/shopping/payment.tpl b/data/Smarty/templates/default/shopping/payment.tpl
index 33c8099..c68f0b5 100644
--- a/data/Smarty/templates/default/shopping/payment.tpl
+++ b/data/Smarty/templates/default/shopping/payment.tpl
@@ -179,6 +179,15 @@
                         </tr>
                     </thead>
                     <tbody>
+                        <!--{if $arrForm[$key].value == "SC_Helper_AmazonPay::PAYMENT_ID"|constant}-->
+                        <tr>
+                            <td class="alignC"><input type="radio" id="pay_1" name="<!--{$key}-->"  value="<!--{$arrForm.payment_id.value|h}-->" style="<!--{$arrErr[$key]|sfGetErrorColor}-->" checked="checked" /></td>
+                            <td>
+                                <label for="pay_1">Amazon Pay</label>
+                            </td>
+                        </tr>
+                        <!--{else}-->
+                        <!--{if $arrPayment[cnt].payment_id != "SC_Helper_AmazonPay::PAYMENT_ID"|constant}-->
                         <!--{section name=cnt loop=$arrPayment}-->
                             <tr>
                             <td class="alignC"><input type="radio" id="pay_<!--{$smarty.section.cnt.iteration}-->" name="<!--{$key}-->"  value="<!--{$arrPayment[cnt].payment_id}-->" style="<!--{$arrErr[$key]|sfGetErrorColor}-->" <!--{$arrPayment[cnt].payment_id|sfGetChecked:$arrForm[$key].value}--> /></td>
@@ -194,6 +203,8 @@
                             <!--{/if}-->
                             </tr>
                         <!--{/section}-->
+                        <!--{/if}-->
+                        <!--{/if}-->
                     </tbody>
                 </table>
             </div>
diff --git a/data/Smarty/templates/sphone/cart/index.tpl b/data/Smarty/templates/sphone/cart/index.tpl
index 6d6859e..66745c8 100644
--- a/data/Smarty/templates/sphone/cart/index.tpl
+++ b/data/Smarty/templates/sphone/cart/index.tpl
@@ -149,6 +149,9 @@
                             <div class="btn_area_btm">
                                 <input type="submit" value="ご購入手続きへ" name="confirm" class="btn data-role-none" />
                             </div>
+                            <div class="btn_area_btm">
+                                <div id="AmazonPayButton<!--{$key}-->" style="text-align: center"></div>
+                            </div>
                         <!--{/if}-->
                     </div><!-- /.formBox -->
                 </form>
@@ -165,4 +168,5 @@
 
 <!--{include file= 'frontparts/search_area.tpl'}-->
 
+<!--{include $tpl_cart_button}-->
 <!--▲コンテンツここまで -->
diff --git a/data/Smarty/templates/sphone/shopping/confirm.tpl b/data/Smarty/templates/sphone/shopping/confirm.tpl
index 1e6943c..8770fd8 100644
--- a/data/Smarty/templates/sphone/shopping/confirm.tpl
+++ b/data/Smarty/templates/sphone/shopping/confirm.tpl
@@ -195,6 +195,7 @@
             <section class="delivconfirm_area">
                 <h3 class="subtitle">お届け先</h3>
 
+                <!--{if !empty($arrForm['memo07'])}--> <p><a href="javascript:;" id="changeButton1"><strong>お届け先を変更する</strong></a></p><!--{/if}-->
                 <div class="form_area">
 
                     <!--{foreach item=shippingItem from=$arrShipping name=shippingItem}-->
@@ -272,7 +273,7 @@
                         <em>配送方法</em>：<!--{$arrDeliv[$arrForm.deliv_id]|h}-->
                     </div>
                     <div class="innerBox">
-                        <em>お支払方法：</em><!--{$arrForm.payment_method|h}-->
+                        <em>お支払方法：</em><!--{if !empty($arrForm['memo07'])}--><!--{$arrForm['memo07']|h}--><br> <a href="javascript:;" id="changeButton2">お支払い方法を変更する</a><!--{else}--><!--{$arrForm.payment_method|h}--> <!--{/if}-->
                     </div>
                     <div class="innerBox">
                         <em>その他お問い合わせ：</em><br />
@@ -299,4 +300,15 @@
 
 <!--{include file= 'frontparts/search_area.tpl'}-->
 
+<script src="https://static-fe.payments-amazon.com/checkout.js"></script>
+<script type="text/javascript" charset="utf-8">
+  amazon.Pay.bindChangeAction('#changeButton1', {
+     amazonCheckoutSessionId: '<!--{$arrForm['memo04']|h}-->',
+     changeAction: 'changeAddress'
+ });
+ amazon.Pay.bindChangeAction('#changeButton2', {
+     amazonCheckoutSessionId: '<!--{$arrForm['memo04']|h}-->',
+     changeAction: 'changePayment'
+ });
+</script>
 <!--▲コンテンツここまで -->
diff --git a/data/Smarty/templates/sphone/shopping/payment.tpl b/data/Smarty/templates/sphone/shopping/payment.tpl
index 6abe5d5..f6e1863 100644
--- a/data/Smarty/templates/sphone/shopping/payment.tpl
+++ b/data/Smarty/templates/sphone/shopping/payment.tpl
@@ -157,6 +157,15 @@
             <!--{/if}-->
             <p class="non-select-msg information">まずはじめに、配送方法を選択ください。</p>
             <ul id="payment">
+                <!--{if $arrForm[$key].value == "SC_Helper_AmazonPay::PAYMENT_ID"|constant}-->
+                        <tr>
+                            <td class="alignC"><input type="radio" id="pay_1" name="<!--{$key}-->"  value="<!--{$arrForm.payment_id.value|h}-->" style="<!--{$arrErr[$key]|sfGetErrorColor}-->" checked="checked" /></td>
+                            <td>
+                                <label for="pay_1">Amazon Pay</label>
+                            </td>
+                        </tr>
+                        <!--{else}-->
+                        <!--{if $arrPayment[cnt].payment_id != "SC_Helper_AmazonPay::PAYMENT_ID"|constant}-->
                 <!--{section name=cnt loop=$arrPayment}-->
                     <li>
                         <input type="radio" id="pay_<!--{$smarty.section.cnt.iteration}-->" name="<!--{$key}-->" value="<!--{$arrPayment[cnt].payment_id}-->" style="<!--{$arrErr[$key]|sfGetErrorColor}-->" <!--{$arrPayment[cnt].payment_id|sfGetChecked:$arrForm[$key].value}--> class="data-role-none" />
@@ -168,6 +177,8 @@
                         <!--{/if}-->
                     </li>
                 <!--{/section}-->
+                <!--{/if}-->
+                <!--{/if}-->
             </ul>
         </section>
 
diff --git a/data/class/SC_Display.php b/data/class/SC_Display.php
index 86ae9e9..f137571 100644
--- a/data/class/SC_Display.php
+++ b/data/class/SC_Display.php
@@ -77,6 +77,7 @@ class SC_Display
         }
         $this->assignobj($page);
         $this->view->setPage($page);
+
         $this->response->setResposeBody($this->view->getResponse($page->getTemplate()));
     }
 
diff --git a/data/class/pages/admin/order/LC_Page_Admin_Order_Edit.php b/data/class/pages/admin/order/LC_Page_Admin_Order_Edit.php
index b7c80c3..7ff26f9 100644
--- a/data/class/pages/admin/order/LC_Page_Admin_Order_Edit.php
+++ b/data/class/pages/admin/order/LC_Page_Admin_Order_Edit.php
@@ -393,6 +393,7 @@ class LC_Page_Admin_Order_Edit extends LC_Page_Admin_Order_Ex
         $objFormParam->addParam('お支払い方法', 'payment_id', INT_LEN, 'n', array('EXIST_CHECK', 'MAX_LENGTH_CHECK', 'NUM_CHECK'));
         $objFormParam->addParam('対応状況', 'status', INT_LEN, 'n', array('EXIST_CHECK', 'MAX_LENGTH_CHECK', 'NUM_CHECK'));
         $objFormParam->addParam('お支払方法名称', 'payment_method');
+        $objFormParam->addParam('AmazonリファレンスID', 'memo05');
 
         // 受注詳細情報
         $objFormParam->addParam('商品種別ID', 'product_type_id', INT_LEN, 'n', array('EXIST_CHECK', 'MAX_LENGTH_CHECK', 'NUM_CHECK'), '0');
diff --git a/data/class/pages/cart/LC_Page_Cart.php b/data/class/pages/cart/LC_Page_Cart.php
index 102ab10..9f5bbe4 100644
--- a/data/class/pages/cart/LC_Page_Cart.php
+++ b/data/class/pages/cart/LC_Page_Cart.php
@@ -215,6 +215,17 @@ class LC_Page_Cart extends LC_Page_Ex
 
         // 全てのカートの内容を取得する
         $this->cartItems = $objCartSess->getAllCartList();
+
+        if (isset($_SESSION['amazonpay_error'])) {
+            $this->tpl_message = 'Amazon Pay のお支払い中にエラーが発生しました。恐れ入りますが、別のお支払い方法をご利用ください。';
+            unset($_SESSION['amazonpay_error']);
+        }
+
+        $amazonpay_config = SC_Helper_AmazonPay::getConfig();
+        $objAmazonPay = new SC_Helper_AmazonPay();
+        $this->payload = SC_Helper_AmazonPay::generatePayloadByCart($this->cartKeys);
+        $this->signature = $objAmazonPay->generateSignatureByCart($this->payload);
+        $this->tpl_cart_button = SC_Helper_AmazonPay::getTemplatePath('cart-button');
     }
 
     /**
diff --git a/data/class/pages/shopping/LC_Page_Shopping.php b/data/class/pages/shopping/LC_Page_Shopping.php
index 0ebf40b..4cbf74c 100644
--- a/data/class/pages/shopping/LC_Page_Shopping.php
+++ b/data/class/pages/shopping/LC_Page_Shopping.php
@@ -248,6 +248,23 @@ class LC_Page_Shopping extends LC_Page_Ex
                 } else {
                     // 通常はログインページ
                     $this->lfInitLoginFormParam($objFormParam);
+
+                    $amazonpay_config = SC_Helper_AmazonPay::getConfig();
+                    $objAmazonPay = new SC_Helper_AmazonPay();
+                    $this->payload = SC_Helper_AmazonPay::generatePayloadByLogin(SHOPPING_URL);
+                    $this->signature = $objAmazonPay->generateSignature($this->payload);
+
+                    if (!empty($_GET['buyerToken'])) {
+                        $buyer = $objAmazonPay->getBuyer($_GET['buyerToken']);
+                        $objAmazonPay->registerCustomer($buyer, $buyer['buyerId']);
+                        $objCustomer->setLogin($buyer['email']);
+
+                        SC_Response_Ex::sendRedirect(
+                            $this->getNextLocation($this->cartKey, $this->tpl_uniqid,
+                                                   $objCustomer, $objPurchase,
+                                                   $objSiteSess));
+                        SC_Response_Ex::actionExit();
+                    }
                 }
 
                 $this->setFormParams($objFormParam, $objPurchase, $this->tpl_uniqid);
diff --git a/data/class/pages/shopping/LC_Page_Shopping_Confirm.php b/data/class/pages/shopping/LC_Page_Shopping_Confirm.php
index eacb1bc..da07734 100644
--- a/data/class/pages/shopping/LC_Page_Shopping_Confirm.php
+++ b/data/class/pages/shopping/LC_Page_Shopping_Confirm.php
@@ -85,6 +85,7 @@ class LC_Page_Shopping_Confirm extends LC_Page_Ex
 
         $this->is_multiple = $objPurchase->isMultiple();
 
+
         // 前のページで正しく登録手続きが行われた記録があるか判定
         if (!$objSiteSess->isPrePage()) {
             // エラー時は、正当なページ遷移とは認めない
@@ -139,6 +140,7 @@ class LC_Page_Shopping_Confirm extends LC_Page_Ex
 
         // 決済モジュールを使用するかどうか
         $this->use_module = SC_Helper_Payment_Ex::useModule($this->arrForm['payment_id']);
+        $arrPayment = SC_Helper_Payment_Ex::get($this->arrForm['payment_id']);
 
         switch ($this->getMode()) {
             // 前のページに戻る
@@ -166,6 +168,9 @@ class LC_Page_Shopping_Confirm extends LC_Page_Ex
                     $objPurchase->completeOrder(ORDER_PENDING);
 
                     SC_Response_Ex::sendRedirect(SHOPPING_MODULE_URLPATH);
+                } elseif ($arrPayment['module_code'] == SC_Helper_AmazonPay::MODULE_CODE) {
+                    $objPurchase->completeOrder(ORDER_PENDING);
+                    SC_Response_Ex::sendRedirect('/shopping/amazonpay.php', ['mode' => 'updateCheckout']);
                 // 購入完了ページ
                 } else {
                     $objPurchase->completeOrder(ORDER_NEW);
@@ -179,5 +184,15 @@ class LC_Page_Shopping_Confirm extends LC_Page_Ex
                 break;
         }
 
+        if ($this->arrForm['payment_id'] == SC_Helper_AmazonPay::PAYMENT_ID && empty($this->arrForm['memo07'])) {
+            $this->renderAmazonPayButton = true;
+            $this->key = $this->cartKey;
+            $this->cartItems = [$this->cartKey => 1];
+            $amazonpay_config = SC_Helper_AmazonPay::getConfig();
+            $objAmazonPay = new SC_Helper_AmazonPay();
+            $this->payload = SC_Helper_AmazonPay::generatePayloadByCart([$this->cartKey]);
+            $this->signature = $objAmazonPay->generateSignatureByCart($this->payload);
+            $this->tpl_cart_button = SC_Helper_AmazonPay::getTemplatePath('cart-button');
+        }
     }
 }
diff --git a/html/shopping/amazonpay.php b/html/shopping/amazonpay.php
new file mode 100755
index 0000000..36d4946
--- /dev/null
+++ b/html/shopping/amazonpay.php
@@ -0,0 +1,6 @@
+<?php
+require_once '../require.php';
+
+$objPage = new LC_Page_Shopping_AmazonPay();
+$objPage->init();
+$objPage->process();
