diff --git a/data/Smarty/templates/default/cart/index.tpl b/data/Smarty/templates/default/cart/index.tpl
index 657b889..6a7c187 100644
--- a/data/Smarty/templates/default/cart/index.tpl
+++ b/data/Smarty/templates/default/cart/index.tpl
@@ -167,6 +167,9 @@
                                         <input type="image" class="hover_change_image" src="<!--{$TPL_URLPATH|h}-->img/button/btn_buystep.jpg" alt="購入手続きへ" name="confirm" />
                                     <!--{/if}-->
                                 </li>
+                                <li style="display:block; width:100%; padding-right: 0">
+                                    <div style="width:400px;height:45px; margin:10px auto" id="AmazonPayButton<!--{$key}-->"></div>
+                                </li>
                             </ul>
                         </div>
                     </form>
@@ -177,3 +180,4 @@
         <!--{/if}-->
     </div>
 </div>
+<!--{include $tpl_cart_button}-->
diff --git a/data/Smarty/templates/default/shopping/payment.tpl b/data/Smarty/templates/default/shopping/payment.tpl
index 33c8099..c68f0b5 100644
--- a/data/Smarty/templates/default/shopping/payment.tpl
+++ b/data/Smarty/templates/default/shopping/payment.tpl
@@ -179,6 +179,15 @@
                         </tr>
                     </thead>
                     <tbody>
+                        <!--{if $arrForm[$key].value == "SC_Helper_AmazonPay::PAYMENT_ID"|constant}-->
+                        <tr>
+                            <td class="alignC"><input type="radio" id="pay_1" name="<!--{$key}-->"  value="<!--{$arrForm.payment_id.value|h}-->" style="<!--{$arrErr[$key]|sfGetErrorColor}-->" checked="checked" /></td>
+                            <td>
+                                <label for="pay_1">Amazon Pay</label>
+                            </td>
+                        </tr>
+                        <!--{else}-->
+                        <!--{if $arrPayment[cnt].payment_id != "SC_Helper_AmazonPay::PAYMENT_ID"|constant}-->
                         <!--{section name=cnt loop=$arrPayment}-->
                             <tr>
                             <td class="alignC"><input type="radio" id="pay_<!--{$smarty.section.cnt.iteration}-->" name="<!--{$key}-->"  value="<!--{$arrPayment[cnt].payment_id}-->" style="<!--{$arrErr[$key]|sfGetErrorColor}-->" <!--{$arrPayment[cnt].payment_id|sfGetChecked:$arrForm[$key].value}--> /></td>
@@ -194,6 +203,8 @@
                             <!--{/if}-->
                             </tr>
                         <!--{/section}-->
+                        <!--{/if}-->
+                        <!--{/if}-->
                     </tbody>
                 </table>
             </div>
diff --git a/data/class/pages/cart/LC_Page_Cart.php b/data/class/pages/cart/LC_Page_Cart.php
index 102ab10..9f5bbe4 100644
--- a/data/class/pages/cart/LC_Page_Cart.php
+++ b/data/class/pages/cart/LC_Page_Cart.php
@@ -215,6 +215,17 @@ class LC_Page_Cart extends LC_Page_Ex
 
         // 全てのカートの内容を取得する
         $this->cartItems = $objCartSess->getAllCartList();
+
+        if (isset($_SESSION['amazonpay_error'])) {
+            $this->tpl_message = 'Amazon Pay のお支払い中にエラーが発生しました。恐れ入りますが、別のお支払い方法をご利用ください。';
+            unset($_SESSION['amazonpay_error']);
+        }
+
+        $amazonpay_config = SC_Helper_AmazonPay::getConfig();
+        $objAmazonPay = new SC_Helper_AmazonPay();
+        $this->payload = SC_Helper_AmazonPay::generatePayloadByCart($this->cartKeys);
+        $this->signature = $objAmazonPay->generateSignatureByCart($this->payload);
+        $this->tpl_cart_button = SC_Helper_AmazonPay::getTemplatePath('cart-button');
     }
 
     /**
diff --git a/data/class/pages/shopping/LC_Page_Shopping_Confirm.php b/data/class/pages/shopping/LC_Page_Shopping_Confirm.php
index eacb1bc..244a06d 100644
--- a/data/class/pages/shopping/LC_Page_Shopping_Confirm.php
+++ b/data/class/pages/shopping/LC_Page_Shopping_Confirm.php
@@ -139,6 +139,7 @@ class LC_Page_Shopping_Confirm extends LC_Page_Ex
 
         // 決済モジュールを使用するかどうか
         $this->use_module = SC_Helper_Payment_Ex::useModule($this->arrForm['payment_id']);
+        $arrPayment = SC_Helper_Payment_Ex::get($this->arrForm['payment_id']);
 
         switch ($this->getMode()) {
             // 前のページに戻る
@@ -166,6 +167,9 @@ class LC_Page_Shopping_Confirm extends LC_Page_Ex
                     $objPurchase->completeOrder(ORDER_PENDING);
 
                     SC_Response_Ex::sendRedirect(SHOPPING_MODULE_URLPATH);
+                } elseif ($arrPayment['module_code'] == SC_Helper_AmazonPay::MODULE_CODE) {
+                    $objPurchase->completeOrder(ORDER_PENDING);
+                    SC_Response_Ex::sendRedirect('/shopping/amazonpay.php', ['mode' => 'updateCheckout']);
                 // 購入完了ページ
                 } else {
                     $objPurchase->completeOrder(ORDER_NEW);
diff --git a/html/shopping/amazonpay.php b/html/shopping/amazonpay.php
new file mode 100755
index 0000000..36d4946
--- /dev/null
+++ b/html/shopping/amazonpay.php
@@ -0,0 +1,6 @@
+<?php
+require_once '../require.php';
+
+$objPage = new LC_Page_Shopping_AmazonPay();
+$objPage->init();
+$objPage->process();
